File: system-test/dotnet/Dsl.Core/UseCase/External/Tax/UseCases/Base/BaseTaxUseCase.cs
using Driver.Port.External.Tax;
using Driver.Port.External.Tax.Dtos.Error;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Tax.UseCases.Base;

public abstract class BaseTaxUseCase<TResponse, TVerification>
    where TVerification : ResponseVerification<TResponse>
{
    protected readonly ITaxDriver _driver;
    protected readonly UseCaseContext _context;

    protected BaseTaxUseCase(ITaxDriver driver, UseCaseContext context)
    {
        _driver = driver;
        _context = context;
    }

    public abstract Task<TaxUseCaseResult<TResponse, TVerification>> Execute();
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/external/tax/usecases/base/BaseTaxUseCase.java
package com.mycompany.myshop.testkit.dsl.core.usecase.external.tax.usecases.base;

import com.mycompany.myshop.testkit.dsl.core.shared.BaseUseCase;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseContext;
import com.mycompany.myshop.testkit.driver.port.external.tax.TaxDriver;

public abstract class BaseTaxUseCase<TResponse, TVerification> extends BaseUseCase<TaxDriver, TResponse, TVerification> {
    protected BaseTaxUseCase(TaxDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}
